core/objlist.hpp

```diff
@@ -307,6 +310,10 @@ class ObjList : public ObjListBase {
         return ret;
     }
 
+    virtual void clear_page_from_memory(Page* page) {
+
```

Comment:
yes, implementation of this function will change the objlist a lot. I consider it may be better to put it into a separate pull request.